contracts/staking/StakingPool.sol

```diff
@@ -602,5 +602,8 @@ contract StakingPool is
             );
             _supportedRewards[address(rewardPools[i].tokens)] = true;
         }
+        for (uint256 i = 0; i < rewardPools.length; i++) {
```

Comment:
question: why not simply remove the setting as true?

line 603 sets the mapping value to true, then this added loop sets them all to false.

Reply:
because then the function wouldnt function as expected?

consider `rewardPools = [{tokens: BITAddress}, {tokens: BITAddress}]` , with line 603 removed, the require statement would pass for i=0 and i=1, which isn't the expected behaviour from the function?

Reply:
Right ...okay,  I think I understand what and how this function is doing it.

You're emptying the class field `_supportedRewards`, to be able to use on future iterations (resetting the state).